Description: Accept mail for postmaster
 Accept mail destined to postmaster as required by RFC5321 ยง4.5.1.
Origin: upstream
Bug: https://github.com/poolpOrg/OpenSMTPD/issues/289
Applied-Upstream: https://github.com/poolpOrg/OpenSMTPD/commit/5e538842
---
This patch header follows DEP-3: http://dep.debian.net/deps/dep3/
Index: opensmtpd/smtpd/smtp_session.c
===================================================================
--- opensmtpd.orig/smtpd/smtp_session.c	2013-08-17 12:18:07.746508316 -0400
+++ opensmtpd/smtpd/smtp_session.c	2013-08-17 12:18:07.746508316 -0400
@@ -1517,6 +1517,16 @@
 		    maddr->user[0] == '\0' &&
 		    maddr->domain[0] == '\0')
 			return (1);
+
+		/* We accept empty domain for RCPT TO if user is postmaster */
+		if (!mailfrom &&
+		    strcasecmp(maddr->user, "postmaster") == 0 &&
+		    maddr->domain[0] == '\0') {
+			(void)strlcpy(maddr->domain, env->sc_hostname,
+			    sizeof maddr->domain);
+			return (1);
+		}
+
 		return (0);
 	}
 
